### PR TITLE
Parameterize truncation when deparsing vectors and lists

### PR DESCRIPTION
**Pull request by ianmcook**
I maintain the [tidyquery](https://github.com/ianmcook/tidyquery) package which uses `rlang::expr_deparse()` to power the `show_dplyr()` function. I would like to be able to disable the truncation of deparsed vectors and lists. This PR adds a new argument `max_len` (edit: renamed `max_elements`) to control this. Thank you!

### CODE REVIEW

File: R/deparse.R [hunk 1]
```diff
@@ -655,9 +655,9 @@ list_deparse <- function(x, lines = new_lines()) {
   lines$push(paste0("<list: "))
   lines$increase_indent()
 
-  truncated <- length(x) > 5L
+  truncated <- !is.null(max_len) && length(x) > max_len
   if (truncated) {
-    x <- .subset(x, 1:5)
+    x <- .subset(x, 1:max_len)
```
Comment (lionel-) [inline R/deparse.R]: Should be `seq_len()` in case `max_len` is 0.

File: R/deparse.R [hunk 2]
```diff
@@ -703,7 +703,7 @@ default_deparse <- function(x, lines = new_lines()) {
   lines$get_lines()
 }
 
-sexp_deparse <- function(x, lines = new_lines()) {
+sexp_deparse <- function(x, lines = new_lines(), max_len = 5L) {
```
Comment (lionel-) [inline R/deparse.R]: This function should validate `max_len`.

File: R/expr.R [hunk 3]
```diff
@@ -354,6 +354,8 @@ switch_expr <- function(.x, ...) {
 #' @param x An object or expression to print.
 #' @param width The width of the deparsed or printed expression.
 #'   Defaults to the global option `width`.
+#' @param max_len Maximum length of a vector or list before truncation
```
Comment (lionel-) [inline R/expr.R]: Probably better as `max_elements`?

File: tests/testthat/test-deparse.R [hunk 4]
```diff
@@ -250,11 +250,16 @@ test_that("literal lists are deparsed", {
   expect_identical(sexp_deparse(list(TRUE, b = 2L, 3, d = "4", as.raw(5))), "<list: TRUE, b = 2L, 3, d = \"4\", <raw: 05>>")
 })
 
-test_that("long vectors are truncated", {
+test_that("long vectors are truncated by default", {
   expect_identical(sexp_deparse(1:10), "<int: 1L, 2L, 3L, 4L, 5L, ...>")
   expect_identical(sexp_deparse(as.list(1:10)), "<list: 1L, 2L, 3L, 4L, 5L, ...>")
 })
 
+test_that("long vectors are not truncated when max_len = NULL", {
```
Comment (lionel-) [inline tests/testthat/test-deparse.R]: Can you add a test for the 0 boundary case please?

File: R/deparse.R (hunk 1 above)
Comment (ianmcook) [reply R/deparse.R]: Done

File: R/deparse.R (hunk 2 above)
Comment (ianmcook) [reply R/deparse.R]: Done

File: R/expr.R (hunk 3 above)
Comment (ianmcook) [reply R/expr.R]: Renamed

File: tests/testthat/test-deparse.R (hunk 4 above)
Comment (ianmcook) [reply tests/testthat/test-deparse.R]: Done (and it didn't work correctly in that case, so I fixed it)

**Comment by ianmcook** [comment]
@lionel- any chance this could make it into the next release? Thank you

**Comment by lionel-** [comment]
I would like to but it makes me _so_ nervous to add a single argument without a complete design of desired options and a consistent naming scheme for all of these :)

I'll keep your PR in mind for the next release, thanks for reminding me.

**Comment by lionel-** [comment]
Thanks! Sorry for the delay in reviewing this.

**Comment by ianmcook** [comment]
@lionel- thank you! The `NEWS.md` bullet ended up in the wrong place, at the bottom of the list for *rlang 0.4.3*. Could you move it under *rlang (development version)* please? Thanks

**Comment by lionel-** [comment]
oops, I did this already but it looks like that news bullet really want to be in 0.4.3
